aiida/engine/processes/functions.py

```diff
@@ -461,7 +462,7 @@ def _define(cls, spec):  # pylint: disable=unused-argument
             spec.inputs.help = namespace_help_string
 
             # If the function supports varargs or kwargs then allow dynamic inputs, otherwise disallow
-            spec.inputs.dynamic = keywords is not None or varargs
+            spec.inputs.dynamic = var_positional or var_keyword
```

Comment:
`dynamic` is a bool type
```suggestion
            spec.inputs.dynamic = var_positional is not None or var_keyword is not None
```

Reply:
Good catch. I am a bit surprised that this didn't cause problems, but apparently it is allowed to set it to `None
```python
In [1]: from aiida.engine import Process

In [2]: Process.spec().inputs.dynamic
Out[2]: False

In [3]: Process.spec().inputs.dynamic = None
```